src/codeu/chat/server/Server.java

```diff
@@ -57,19 +61,32 @@
   private final Uuid id;
   private final Secret secret;
 
-  private final Model model = new Model();
-  private final View view = new View(model);
   private final Controller controller;
 
   private final Relay relay;
   private Uuid lastSeen = Uuid.NULL;
 
-  public Server(final Uuid id, final Secret secret, final Relay relay) {
+  private final Model model;
+  final View view;
+
+  public Server(final Uuid id, final Secret secret, final Relay relay, final File dir) {
 
     this.id = id;
     this.secret = secret;
-    this.controller = new Controller(id, model);
     this.relay = relay;
+    File fp = new File(dir, "chatLog.dat");
+    if (fp == null) {
```

Comment:
You get a FileNotFoundException when you load the file isn't there. You need to check if fp exists too with File.exists:

https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#exists-java.nio.file.Path-java.nio.file.LinkOption...-